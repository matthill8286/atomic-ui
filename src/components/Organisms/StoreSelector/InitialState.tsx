/* global JSX */
import React from 'react'
import { IconAddToPlaylist } from '@matthill8286/atomic-icon-library'
import { InitialStateWrapper, CopyTextCentralized, SVGCentralized } from './StoreSelector.styled'
// eslint-disable-next-line prettier/prettier
import type { InitialStateProps } from './StoreSelector.types'

export function InitialState({ text, children }: InitialStateProps): JSX.Element {
  return (
    <InitialStateWrapper>
      <SVGCentralized width={108} height={108}>
        <IconAddToPlaylist />
      </SVGCentralized>
      <CopyTextCentralized fontSize="md" weight="semibold" tag="p" textAlign="center">
        {text}
      </CopyTextCentralized>
      {children}
    </InitialStateWrapper>
  )
}
