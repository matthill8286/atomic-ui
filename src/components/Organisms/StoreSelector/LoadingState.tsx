/* global JSX */
import React from 'react'
import { SkeletonBlockItem } from '@/components/Atoms/Skeleton'

export function LoadingState(): JSX.Element {
  return (
    <>
      {[1, 2].map((i) => (
        <div key={i} style={{ marginTop: '24px' }} data-test="store-selector-loading-state">
          <SkeletonBlockItem width="50%" height="25px" />
          <SkeletonBlockItem width="80%" height="50px" marginTop="16px" />
          <SkeletonBlockItem width="100%" height="1px" marginTop="16px" />
        </div>
      ))}
    </>
  )
}
