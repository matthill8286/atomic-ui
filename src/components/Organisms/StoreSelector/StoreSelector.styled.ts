import { Button } from '@/components/Atoms/Button'
import { CopyText } from '@/components/Atoms/Typography'
import { Icon } from '@/components/Atoms/Icon'
import { Input } from '@/components/Atoms/Input'
import { media } from '@/styles/media'
import { css, styled } from '@/styles/styled'

export const FlexItem = styled.div<{ display?: string; align?: string; maxWidth?: string }>`
  flex: 1 1 auto;
  flex-direction: column;
  display: ${({ display }) => display || 'inherit'};
  align-items: flex-start;
  ${media.md} {
    align-items: ${({ align }) => align || 'inherit'};
    max-width: ${({ maxWidth }) => maxWidth || 'auto'};
  }
`

export const CopyTextCentralized = styled(CopyText)`
  margin-left: auto;
  margin-right: auto;
`

export const SVGCentralized = styled(Icon)`
  & svg {
    margin-left: auto;
    margin-right: auto;
  }
`

export const InitialStateWrapper = styled.div`
  max-width: 390px;
  margin-left: auto;
  margin-right: auto;
`

export const MainWrapper = styled.div<{ fullWidth?: boolean }>`
  max-width: ${({ fullWidth }) => (fullWidth ? '100%' : '400px')};
  margin: 0 auto;
  padding: ${({ theme }) => theme.spacing.base.xs} 0;
  ${media.md} {
    max-width: ${({ fullWidth }) => (fullWidth ? '100%' : '650px')};
  }
`

export const ItemWrapper = styled.div`
  display: flex;
  flex-wrap: wrap;
  padding: ${({ theme }) => `${theme.spacing.base.xl} 0`};
  border-bottom: 1px solid ${({ theme }) => theme.color.grey2};

  &:last-child {
    border-bottom: none;
  }
`

export const Badge = styled.span(
  ({ theme }) => css`
    position: relative;
    top: -3px;
    margin-right: ${theme.spacing.base.xs};
    padding: 0 ${theme.spacing.base.xs};
    background: ${theme.color.primary};
    border-radius: ${theme.dimension.borderRadius2};
    color: ${theme.color.white};
    font-family: ${theme.font.family.default};
    font-size: ${theme.font.size.xxs};
  `
)

export const SInput = styled(Input)`
  & input {
    &::placeholder {
      opacity: 1;
    }
  }

  ${Icon} {
    cursor: pointer;
  }
`

export const SSearchWrapper = styled.div<{ hasConsent: boolean }>`
  position: relative;
  min-height: ${({ hasConsent }) => (hasConsent ? '278px' : 'auto')};
`

export const ActionButton = styled(Button)`
  max-width: inherit;
  width: 100%;

  ${media.md} {
    width: auto;
    max-width: 300px;
    margin: 0;
  }
`
