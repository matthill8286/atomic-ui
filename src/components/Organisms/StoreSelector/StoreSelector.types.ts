/* global React */
export interface Item {
  id: string
  badge?: string
  header: string
  description?: React.ReactNode | null
  [key: string]: unknown
}

export type SearchFieldProps = {
  consent?: React.ReactNode
  disabled?: boolean
  errorMessage?: string
  onChange: (value: string) => void
  onSearch: () => void
  placeholder: string
  value: string
}

export type InitialStateProps = {
  text: string
  children: React.ReactNode
}

export type StoreSelectorProps = {
  componentComplement?: <T extends Item>(item: T) => React.ReactNode | void
  componentSearchConsent?: React.ReactNode
  hideCloseButton?: boolean
  initialSearchQuery?: string
  isDisabledPredicate?: <T extends Item>(item: T) => boolean
  isLoading?: boolean
  isSelectedPredicate?: <T extends Item>(item: T) => boolean
  itemList?: Item[]
  itemSelected?: Item | null
  noModal?: boolean
  onModalClose?: () => void
  onSearch: (query: string) => void
  onSelect: (item: Item) => void
  textErrorMessage: string
  textInitialState: string
  textModalTitle: string
  textNoItemsFound: string
  textSearchPlaceholder: string
  textSelectDelivery: string
  textSelected: string
  textSelectPickup: string
}
