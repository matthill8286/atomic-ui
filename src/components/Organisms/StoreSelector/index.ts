export * from './StoreSelector'
export * from './StoreSelector.types'
