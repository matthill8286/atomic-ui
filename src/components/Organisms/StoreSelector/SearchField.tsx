/* global JSX */
import React from 'react'
import { InfoText } from '@/components/Atoms/Typography'
import { IconSearch } from '@matthill8286/atomic-icon-library'
import { SSearchWrapper, SInput } from './StoreSelector.styled'
// eslint-disable-next-line prettier/prettier
import type { SearchFieldProps } from './StoreSelector.types'

export function SearchField({
  placeholder,
  value,
  onChange,
  onSearch,
  consent = null,
  errorMessage,
  disabled,
}: SearchFieldProps): JSX.Element {
  function handleInputChange(e: React.ChangeEvent<HTMLInputElement>) {
    onChange(e.target.value)
  }

  return (
    <SSearchWrapper data-test="mms-market-search-field" hasConsent={Boolean(consent)}>
      <SInput
        autofocus
        inputIcon={!errorMessage ? <IconSearch /> : undefined}
        onChange={handleInputChange}
        onClickIcon={onSearch}
        onKeyDown={(e) => value !== '' && e.key === 'Enter' && onSearch()}
        placeholder={placeholder}
        state={errorMessage ? 'error' : disabled ? 'disabled' : undefined}
        value={value}
      />
      {consent}
      {errorMessage && (
        <InfoText color="primary" data-test="mms-market-search-field-error">
          {errorMessage}
        </InfoText>
      )}
    </SSearchWrapper>
  )
}
