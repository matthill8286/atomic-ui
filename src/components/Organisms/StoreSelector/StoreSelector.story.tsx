/* eslint-disable no-console */
import React from 'react'
import { boolean } from '@storybook/addon-knobs'
import { storiesOf } from '@storybook/react'
import { StoreSelector } from './StoreSelector'
import { NotificationBox, Availability, ProductAvailabilityState } from '@/index'

function sharedKnobs() {
  return {
    isLoading: boolean('Loading state', false),
    noModal: boolean('Without modal', false),
    textErrorMessage: '',
    textInitialState: 'Please enter your postal code or city, to view markets in your area.',
    textModalTitle: 'Modal title',
    textNoItemsFound: 'No items found',
    textSearchPlaceholder: 'Search...',
    textSelectDelivery: 'SelectDelivery',
    textSelectPickup: 'Seleccionar para recoger',
    textSelected: 'Selected',
    onModalClose: () => undefined,
  }
}

storiesOf('Design System/Organisms/StoreSelector', module)
  .add('- initial', () => (
    <StoreSelector
      {...sharedKnobs()}
      onSearch={query => console.log('onSearch', query)}
      onSelect={model => console.log('onSelect', model)}
      itemList={[]}
    />
  ))
  .add('- no items found', () => (
    <StoreSelector
      {...sharedKnobs()}
      onSearch={query => console.log('onSearch', query)}
      onSelect={model => console.log('onSelect', model)}
      itemList={[]}
      initialSearchQuery="Barcelona"
    />
  ))
  .add('- example', () => (
    <StoreSelector
      {...sharedKnobs()}
      onSearch={query => console.log('onSearch', query)}
      onSelect={model => console.log('onSelect', model)}
      isSelectedPredicate={item => item.id === 'clubMarketId'}
      isDisabledPredicate={item => item.id === '2'}
      itemList={[
        {
          id: 'clubMarketId',
          badge: '1km',
          header: 'München-Euroindustriepark',
          description: (
            <>
              Centre Comercial Parc Vallés
              <br />
              Avinguda Tèxtil, 47
              <br />
              08223 Terrassa, Barcelona
              <br />
              Lu-Sa: 09:30 - 21:30h,
            </>
          ),
        },
        {
          id: '2',
          badge: '2km',
          header: 'München-Euroindustriepark 2',
          description: 'Damit Sie schneller Ihren nächsten',
        },
        {
          id: '3',
          badge: '33km',
          header: 'München-Euroindustriepark 3',
          description: 'Damit Sie schneller Ihren nächsten',
        },
        {
          id: '4',
          badge: '112km',
          header: 'München-Euroindustriepark 4',
          description: 'Damit Sie schneller Ihren nächsten',
        },
      ]}
    />
  ))
  .add('- complement component', () => (
    <StoreSelector
      {...sharedKnobs()}
      onSearch={query => console.log('onSearch', query)}
      onSelect={model => console.log('onSelect', model)}
      isSelectedPredicate={item => item.id === 'clubMarketId'}
      isDisabledPredicate={item => item.id === '2'}
      componentComplement={item => (
        <div style={{ marginBottom: '8px' }}>
          <Availability text="Availability text" state={ProductAvailabilityState.AVAILABLE} />
        </div>
      )}
      itemList={[
        {
          id: 'clubMarketId',
          badge: '1km',
          header: 'München-Euroindustriepark',
          description: (
            <>
              Centre Comercial Parc Vallés
              <br />
              Avinguda Tèxtil, 47
              <br />
              08223 Terrassa, Barcelona
              <br />
              Lu-Sa: 09:30 - 21:30h,
            </>
          ),
        },
        {
          id: '2',
          badge: '2km',
          header: 'München-Euroindustriepark 2',
          description: 'Damit Sie schneller Ihren nächsten',
        },
        {
          id: '3',
          badge: '33km',
          header: 'München-Euroindustriepark 3',
          description: 'Damit Sie schneller Ihren nächsten',
        },
        {
          id: '4',
          badge: '112km',
          header: 'München-Euroindustriepark 4',
          description: 'Damit Sie schneller Ihren nächsten',
        },
      ]}
    />
  ))
  .add('- search consent', () => (
    <StoreSelector
      {...sharedKnobs()}
      componentSearchConsent={
        <NotificationBox
          buttonLayout="row"
          buttons={[{ actionBtnLabel: 'Aktivieren', onClick: function noRefCheck() {} }]}
          hasTitleIcon
          title="Damit Sie schneller Ihren nächsten Markt finden, nutzen wir Daten von Google Maps. Um fortzufahren müssen Sie die Komfort-Kategorie aktivieren."
          type="info"
          tooltip={{
            arrowPosition: 'top-left',
            left: 0,
            top: 70,
          }}
        />
      }
      onSearch={query => console.log('onSearch', query)}
      onSelect={model => console.log('onSelect', model)}
    />
  ))
