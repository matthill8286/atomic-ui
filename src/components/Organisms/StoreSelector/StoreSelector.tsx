/* global JSX */
/* eslint-disable eqeqeq */
import React, { useState, useRef, useEffect } from 'react'
import anyPass from 'lodash/fp/anyPass'
import { Modal } from '@/components/Molecules/Modal'
import { Heading, Typo } from '@/components/Atoms/Typography'
import { Badge, ActionButton, FlexItem, ItemWrapper, MainWrapper } from './StoreSelector.styled'
// eslint-disable-next-line prettier/prettier
import type { StoreSelectorProps, Item } from './StoreSelector.types'
import { SearchField } from './SearchField'
import { InitialState } from './InitialState'
import { LoadingState } from './LoadingState'

export function StoreSelector({
  componentComplement,
  componentSearchConsent,
  hideCloseButton,
  initialSearchQuery = '',
  isDisabledPredicate,
  isLoading,
  isSelectedPredicate,
  itemList,
  itemSelected,
  noModal,
  onModalClose,
  onSearch,
  onSelect,
  textErrorMessage,
  textInitialState,
  textModalTitle,
  textNoItemsFound,
  textSearchPlaceholder,
  textSelected,
  textSelectPickup,
}: StoreSelectorProps): JSX.Element {
  const isPristineRef = useRef<boolean>(true)
  const [searchQuery, setSearchQuery] = useState(initialSearchQuery)
  const hasItems = Array.isArray(itemList) && itemList.length > 0
  const hasError = textErrorMessage.length > 0
  const isItemSelected = (x: Item) => itemSelected != null && itemSelected?.id === x.id
  const isSelected = isSelectedPredicate
    ? anyPass([isItemSelected, isSelectedPredicate])
    : isItemSelected

  function handleSearch() {
    if (searchQuery !== '') {
      onSearch(searchQuery)
    }
  }

  useEffect(() => {
    if (isPristineRef.current) {
      isPristineRef.current = !(Boolean(isLoading) && Boolean(searchQuery))
    }
  }, [isPristineRef, searchQuery, isLoading])

  const Search = (
    <SearchField
      placeholder={textSearchPlaceholder}
      value={searchQuery}
      onChange={setSearchQuery}
      onSearch={handleSearch}
      consent={componentSearchConsent}
      errorMessage={textErrorMessage}
      disabled={isLoading}
    />
  )

  const WrapperContent = isLoading ? (
    <>
      {Search}
      <LoadingState />
    </>
  ) : !hasItems ? (
    <InitialState text={isPristineRef.current ? textInitialState : textNoItemsFound}>
      {Search}
    </InitialState>
  ) : hasError ? (
    Search
  ) : (
    <>
      {Search}
      {itemList?.map((item) => {
        const isCurrentSelected = isSelected(item)
        const isDisabled = isDisabledPredicate != null && isDisabledPredicate(item)

        return (
          <ItemWrapper key={item.id} data-test="store-selector-item">
            <FlexItem display="block" maxWidth="60%">
              <Badge>{item.badge}</Badge>
              <Heading tag="span" scale="level-4">
                {item.header}
              </Heading>
              <Typo tag="div" fontSize="xs" margin="xxs 0 md 0">
                {item.description}
              </Typo>
            </FlexItem>
            <FlexItem display="flex" align="flex-end">
              {componentComplement != null && componentComplement(item)}
              <ActionButton
                actionType={isCurrentSelected ? 'secondary' : 'primary'}
                disabled={isDisabled}
                onClick={() => onSelect(item)}>
                {isCurrentSelected ? textSelected : textSelectPickup}
              </ActionButton>
            </FlexItem>
          </ItemWrapper>
        )
      })}
    </>
  )

  return noModal ? (
    <MainWrapper fullWidth={noModal}>{WrapperContent}</MainWrapper>
  ) : (
    <Modal title={textModalTitle} onClose={onModalClose} hideCloseButton={hideCloseButton}>
      <MainWrapper>{WrapperContent}</MainWrapper>
    </Modal>
  )
}
