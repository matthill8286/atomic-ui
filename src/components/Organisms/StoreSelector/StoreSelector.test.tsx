import React from 'react'
import { mountWithTheme, renderWithTheme } from '@/testRenderer'
import { StoreSelector } from './StoreSelector'
// eslint-disable-next-line prettier/prettier
import type { StoreSelectorProps } from './StoreSelector.types'

describe('<StoreSelector />', () => {
  const props: StoreSelectorProps = {
    onSearch: jest.fn(),
    onSelect: jest.fn(),
    textErrorMessage: '',
    textInitialState: 'initial message',
    textModalTitle: 'modal title',
    textNoItemsFound: 'no items found',
    textSearchPlaceholder: 'search',
    textSelectDelivery: 'delivery selected',
    textSelected: 'selected',
    textSelectPickup: 'pickup',
  }

  it('renders correct', () => {
    const tree = renderWithTheme(<StoreSelector {...props} />)
    expect(tree).toMatchSnapshot()
  })

  it('Should render without crashing', () => {
    const component = mountWithTheme(<StoreSelector {...props} />)
    expect(component.find(StoreSelector)).toHaveLength(1)
  })

  describe('isLoading property:', () => {
    it('Should render initial state if !isLoading', () => {
      const component = mountWithTheme(<StoreSelector {...props} isLoading={false} />)
      const paragraph = component.find('p')
      expect(paragraph).toHaveLength(1)
      expect(paragraph.text()).toEqual('initial message')
    })

    it('Should not render initial state if isLoading', () => {
      const component = mountWithTheme(<StoreSelector {...props} isLoading />)
      expect(component.find('p')).toHaveLength(0)
    })

    it('Should render loading skeleton if isLoading', () => {
      const component = mountWithTheme(<StoreSelector {...props} isLoading />)
      expect(component.find('div[data-test="store-selector-loading-state"]')).toHaveLength(2)
    })

    it('Should not render loading skeleton if !isLoading', () => {
      const component = mountWithTheme(<StoreSelector {...props} isLoading={false} />)
      expect(component.find('div[data-test="store-selector-loading-state"]')).toHaveLength(0)
    })

    it('Should render items if !isLoading and itemList.length > 0', () => {
      const component = mountWithTheme(
        <StoreSelector
          {...props}
          itemList={[
            {
              id: '1',
              badge: '1km',
              header: 'Header',
              description: 'description',
            },
          ]}
        />
      )
      expect(component.find('div[data-test="store-selector-item"]')).toHaveLength(1)
    })

    it('Should not render items if isLoading is true', () => {
      const component = mountWithTheme(
        <StoreSelector
          {...props}
          isLoading
          itemList={[
            {
              id: '1',
              badge: '1km',
              header: 'Header',
              description: 'description',
            },
          ]}
        />
      )
      expect(component.find('div[data-test="store-selector-item"]')).toHaveLength(0)
    })
  })

  describe('initialSearchQuery property:', () => {
    it('Should render textInitialState text when query is present, results are empty and pristine is true', () => {
      const component = mountWithTheme(
        <StoreSelector {...props} itemList={[]} initialSearchQuery="Barcelona" />
      )
      const paragraph = component.find('p')
      expect(component.find('div[data-test="store-selector-item"]')).toHaveLength(0)
      expect(paragraph).toHaveLength(1)
      expect(paragraph.text()).toEqual('initial message')
    })
  })

  describe('textErrorMessage property:', () => {
    it('Should render error message when error is given', () => {
      const component = mountWithTheme(<StoreSelector {...props} textErrorMessage="ERROR-MSG" />)
      const errorSpan = component.find('span[data-test="mms-market-search-field-error"]')
      expect(errorSpan).toHaveLength(1)
      expect(errorSpan.text()).toEqual('ERROR-MSG')
    })
  })
})
