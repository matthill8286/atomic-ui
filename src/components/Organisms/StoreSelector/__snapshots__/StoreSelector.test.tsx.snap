// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StoreSelector /> renders correct 1`] = `
.c5 {
  margin: 0;
  font-size: 26px;
  line-height: 28px;
  font-family: OswaldLight,Arial,sans-serif;
  font-weight: 600;
  -webkit-letter-spacing: 0.4px;
  -moz-letter-spacing: 0.4px;
  -ms-letter-spacing: 0.4px;
  letter-spacing: 0.4px;
  text-align: center;
}

.c15 {
  color: #666666;
  font-size: 18px;
  line-height: 28px;
  font-family: OswaldLight,Arial,sans-serif;
  font-weight: 600;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  text-align: center;
}

.c21 {
  color: #666666;
  font-size: 16px;
  line-height: 20px;
  font-family: OswaldLight,Arial,sans-serif;
  font-weight: 400;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c7 svg {
  width: 30px;
  height: 30px;
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  fill: #666666;
}

.c7 svg g,
.c7 svg p {
  fill: #666666;
}

.c13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c13 svg {
  width: 108px;
  height: 108px;
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  fill: currentColor;
}

.c13 svg g,
.c13 svg p {
  fill: currentColor;
}

.c27 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c27 svg {
  width: 20px;
  height: 20px;
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  fill: #CCCCCC;
}

.c27 svg g,
.c27 svg p {
  fill: #CCCCCC;
}

.c25 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 4px;
  box-sizing: border-box;
  outline: none;
  cursor: pointer;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  -webkit-appearance: none;
  padding: 12px 24px;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-family: OswaldLight,Arial,sans-serif;
  font-weight: normal;
  font-size: 16px;
  height: 48px;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  box-shadow: none;
  border: 2px solid #00C8DC;
  background: #FFFFFF;
  color: #00C8DC;
}

.c25 .c6 {
  padding: 0 0px;
}

.c25 .Status-y46az8-0 {
  position: absolute;
  top: -6px;
  right: -6px;
}

.c25:hover {
  box-shadow: 0 8px 10px 1px rgba(0,0,0,0.07),0 3px 14px 2px rgba(0,0,0,0.06),0 5px 5px -3px rgba(0,0,0,0.10);
}

.c25 > span {
  color: inherit;
  display: block;
  text-align: center;
  width: 100%;
}

.c26 {
  min-width: 20px;
  height: 20px;
  padding: 0 8px;
}

.c0 {
  z-index: 201;
  top: 0;
  bottom: 0;
  right: 0;
  left: 0;
  position: fixed;
  overflow-y: auto;
}

.c1 {
  position: fixed;
  position: -webkit-sticky;
  position: sticky;
  right: 0;
  bottom: 0;
  top: 0;
  left: 0;
  z-index: 100;
  width: 100%;
  height: 100%;
  background-color: rgba(0,0,0,0.7);
}

.c2 {
  background: #FFFFFF;
  border-radius: 0;
  box-sizing: border-box;
  z-index: 101;
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: column;
  -ms-flex-flow: column;
  flex-flow: column;
  overflow-y: hidden;
  position: absolute;
  top: 0;
}

.c10 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: column;
  -ms-flex-flow: column;
  flex-flow: column;
}

.c10 img {
  max-width: 100%;
}

.c9 {
  padding: 0 16px 16px 16px;
  max-width: 100%;
  overflow-y: scroll;
  -webkit-overflow-scrolling: touch;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 16px;
}

.c8 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  padding: 0;
  z-index: 1;
  outline: none;
  background: none;
  border: none;
}

.c4 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c18 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  vertical-align: top;
  width: 100%;
}

.c18 .c6 {
  position: absolute;
  right: 0;
  top: 28px;
  outline: none;
}

.c18 .c6:focus svg {
  fill: #000000;
}

.c23 {
  border: 0;
  box-sizing: content-box;
  display: block;
  font-size: 16px;
  font-family: OswaldLight,Arial,sans-serif;
  margin: 24px 0 0 0;
  border-bottom: 1px solid #999999;
  background-color: #FFFFFF;
  min-width: 0;
  width: 100%;
  padding: 8px 16px;
  line-height: 40px;
  outline: none;
  position: relative;
  -webkit-tap-highlight-color: transparent;
}

.c23::-webkit-input-placeholder {
  opacity: 1;
}

.c23::-moz-placeholder {
  opacity: 1;
}

.c23:-ms-input-placeholder {
  opacity: 1;
}

.c23::placeholder {
  opacity: 1;
}

.c24 {
  position: absolute;
  top: 16px;
  right: 16px;
  height: 100%;
}

.c24 > button {
  background: none;
  border: none;
  margin: 0;
  padding: 0;
  top: 28px;
  outline: none;
}

.c24 > button:focus svg {
  fill: #000000;
}

.c20 {
  margin-right: 32px;
}

.c22 {
  position: absolute;
}

.c16 {
  margin-left: auto;
  margin-right: auto;
}

.c14 svg {
  margin-left: auto;
  margin-right: auto;
}

.c12 {
  max-width: 390px;
  margin-left: auto;
  margin-right: auto;
}

.c11 {
  max-width: 400px;
  margin: 0 auto;
  padding: 8px 0;
}

.c19 input::-webkit-input-placeholder {
  opacity: 1;
}

.c19 input::-moz-placeholder {
  opacity: 1;
}

.c19 input:-ms-input-placeholder {
  opacity: 1;
}

.c19 input::placeholder {
  opacity: 1;
}

.c19 .c6 {
  cursor: pointer;
}

.c17 {
  position: relative;
  min-height: auto;
}

@media (min-width:512px) {
  .c1 {
    overflow: auto;
    -webkit-overflow-scrolling: touch;
  }
}

@media (min-width:512px) {
  .c2 {
    top: 48px;
    -webkit-transform: translate(-50%);
    -ms-transform: translate(-50%);
    transform: translate(-50%);
    margin-bottom: 64px;
    left: 50%;
    display: block;
    padding: 32px 24px;
    border-radius: 4px;
    height: auto;
    max-width: 756px;
  }
}

@media (min-width:512px) {
  .c9 {
    overflow-y: scroll;
    -webkit-overflow-scrolling: touch;
    padding: 0;
  }
}

@media (min-width:512px) {
  .c3 {
    padding: 0 0 16px 0;
  }
}

@media (min-width:752px) {
  .c11 {
    max-width: 650px;
  }
}

<div
  className="c0"
  id="saiyan-styled-modal-wrapper"
>
  <div
    className="c1"
    onClick={[Function]}
  />
  <div
    className="c2"
    onClick={[Function]}
    size="lg"
  >
    <div
      className="c3"
    >
      <div
        className="c4"
      />
      <div>
        <h1
          className="c5"
          fontFamily="default"
          fontSize="xl"
          spacing="base"
        >
          modal title
        </h1>
      </div>
      <div
        className="c6 c7 c8"
        color="grey5"
        height={30}
        onClick={[Function]}
        width={30}
      >
        <svg
          data-test="modal-close-button"
          height={18}
          width={18}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M10.397 9.02l7.374-7.519A.871.871 0 0016.53.282L9.166 7.79 1.502.27a.87.87 0 10-1.22 1.243l7.653 7.508-7.653 7.507a.871.871 0 001.22 1.243l7.664-7.52 7.363 7.508a.87.87 0 001.242-1.219l-7.374-7.52z"
            fill="#999"
          />
        </svg>
      </div>
    </div>
    <div
      className="c9"
      id="saiyan-styled-modal-desktop-wrapper"
    >
      <div
        className="c10"
      >
        <div
          className="c11"
        >
          <div
            className="c12"
          >
            <div
              className="c6 c13 c14"
              height={108}
              width={108}
            >
              <svg
                data-name="Layer 1"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M20 20a2 2 0 11-2-2 2 2 0 012 2zM8 18a2 2 0 102 2 2 2 0 00-2-2zM21.5 4h-1a.5.5 0 00-.5.5V5a2.86 2.86 0 01-.36 1.37l-2.36 4.13a1 1 0 01-.75.5H10.3a1 1 0 01-.86-.5L5.16 3a2 2 0 00-1.74-1H2.5a.5.5 0 00-.5.5v1a.5.5 0 00.5.5h.92l4.28 7.49a2.19 2.19 0 00.27.35l-1.86 3.7a.77.77 0 000 .71l.22.38A.76.76 0 007 17h12.5a.5.5 0 00.5-.5v-1a.5.5 0 00-.5-.5H8.62l1-2.09a2.74 2.74 0 00.64.09h6.23A2.94 2.94 0 0019 11.49l2.35-4.13A4.76 4.76 0 0022 5v-.5a.5.5 0 00-.5-.5zM16 5.38v-.76a.5.5 0 00-.5-.5h-1.62V2.5a.5.5 0 00-.5-.5h-.76a.5.5 0 00-.5.5v1.62H10.5a.5.5 0 00-.5.5v.76a.5.5 0 00.5.5h1.62V7.5a.5.5 0 00.5.5h.76a.5.5 0 00.5-.5V5.88h1.62a.5.5 0 00.5-.5z"
                />
              </svg>
            </div>
            <p
              className="c15 c16"
              color="grey5"
              fontSize="md"
            >
              initial message
            </p>
            <div
              className="c17"
              data-test="mms-market-search-field"
            >
              <div
                className="c18 c19"
              >
                <div
                  className="c20"
                >
                  <label
                    className="c21 c22"
                    color="grey5"
                    fontSize="sm"
                  >
                    
                  </label>
                  <input
                    aria-invalid={false}
                    aria-required={false}
                    autoFocus={true}
                    className="c23"
                    disabled={false}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                    placeholder="search"
                    required={false}
                    type="text"
                    value=""
                  />
                </div>
                <div
                  className="c24"
                >
                  <button
                    className="c25 c26"
                    disabled={false}
                    height={20}
                    onClick={[Function]}
                    size="md"
                    type="button"
                    width={20}
                  >
                    <span
                      className="c6 c27"
                      color="grey3"
                      height={20}
                      width={20}
                    >
                      <svg
                        viewBox="0 0 24 24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M20.85 19.44l-4-4a2.44 2.44 0 00-.43-.35l-1-.69A7 7 0 1010 17a7 7 0 004.37-1.53l.75 1a2.6 2.6 0 00.3.36l4 4a.5.5 0 00.71 0l.7-.7a.5.5 0 00.02-.69zM10 15a5 5 0 115-5 5 5 0 01-5 5z"
                        />
                      </svg>
                    </span>
                  </button>
                </div>
              </div>
              
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
